velero-plugins/nonadmin/restore.go

```diff
@@ -0,0 +1,55 @@
+package nonadmin
+
+import (
+	"github.com/sirupsen/logrus"
+	"github.com/vmware-tanzu/velero/pkg/plugin/velero"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"strings"
+)
+
+type RestorePluginNonAdmin struct {
+	Log logrus.FieldLogger
+}
+
+const (
+	// API Group
+	GroupOADP = "oadp.openshift.io"
+
+	// Resource Kinds
+	KindNonAdminBackup                = "NonAdminBackup"
+	KindNonAdminRestore               = "NonAdminRestore"
+	KindNonAdminBackupStorageLocation = "NonAdminBackupStorageLocation"
+)
+
+var includedResources = []string{
+	"nonadminbackups.oadp.openshift.io",
+	"nonadminrestores.oadp.openshift.io",
+	"nonadminbackupstoragelocations.oadp.openshift.io",
+}
+
+func (p *RestorePluginNonAdmin) AppliesTo() (velero.ResourceSelector, error) {
+	p.Log.Info("[nonadmin-restore] Applying to NonAdmin resources")
+	return velero.ResourceSelector{
+		IncludedResources: includedResources,
+	}, nil
+}
+
+func (p *RestorePluginNonAdmin) Execute(input *velero.RestoreItemActionExecuteInput) (*velero.RestoreItemActionExecuteOutput, error) {
+	obj, ok := input.Item.(*unstructured.Unstructured)
+	if !ok {
+		p.Log.Warn("[nonadmin-restore] Failed to cast item to Unstructured, skipping")
+		return velero.NewRestoreItemActionExecuteOutput(input.Item), nil
+	}
+
+	apiVersion := obj.GetAPIVersion()
+	kind := obj.GetKind()
+	group := strings.Split(apiVersion, "/")[0]
```

Comment:
Is there a chance the apiVersion will return string without the `/` ?

It should:
```go
oadp.openshift.io/v1alpha1
```

but is there a chance to get only `v1alpha1` using this call?

Reply:
never mind, it will work anyway just group won't match so this won't apply and I think there is always `/` in the api string for a CRD.